FreeRTOS/FreeRTOS-Plus-TCP
# DriverSAM/NetworkInterface.c warning cleanup - purely refactoring

DriverSAM/NetworkInterface.c warning cleanup refactor

Description
-----------

- Removed 6 unused variables to address [-Wunused-variable] warnings. One of these, `uxLowestSemCount` was also shadowing a file scope variable, which is misleading to a reader.
- Created function prototypes for 2 functions to address [-Wmissing-declarations]

Test Steps
-----------
The unused variables did not appear anywhere else in the library and are optimized out when compiling with -O1. With -O1 there is no difference with or without the unused variables.

The function prototypes do not appear in other files, so they are not violating MISRA C 2012 rule 8.5, and adding them satisfies rule 8.4. 

No new unit tests or modifications needed, this is purely a refactor. I have tested functionality within my application [here](https://github.com/Peter-Herrmann/SAME70-FreeRTOS-Plus-TCP).

Checklist:
----------

Related Issue
-----------
No issue related


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
